[breakage] Compile-time typing issues

src/conjugate.ts:83:19 - error TS2322: Type 'unknown[]' is not assignable to type '{ [K in keyof CMixins]: CMixins[K] extends IClass<infer T extends unknown, never[]> ? T : never; }'.

83             const mixins: {
                     ~~~~~~

src/conjugate.ts:98:52 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'object'.

98                 for (const prop of Reflect.ownKeys(mixin)) {
                                                      ~~~~~

src/conjugate.ts:101:41 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Record<string | symbol, unknown>'.

101                         _delegate(prop, mixin, owner);
                                            ~~~~~

src/conjugate.ts:121:41 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'object'.

121                         if (Reflect.has(mixins[i], prop)) return true;
                                            ~~~~~~~~~

src/conjugate.ts:136:41 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'object'.

136                         if (Reflect.has(mixins[i], prop)) {
                                            ~~~~~~~~~

src/conjugate.ts:137:55 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'object'.

137                             const value = Reflect.get(mixins[i], prop, receiver);
                                                          ~~~~~~~~~


Found 6 errors in the same file, starting at: src/conjugate.ts:83

--- a/src/conjugate.ts
+++ b/src/conjugate.ts
@@ -80,9 +80,7 @@ function Conjugate<
 
             // Instantiate base and mixins, save their instances for later use.
             const base = Reflect.construct(Base, argsBase, self.constructor);
-            const mixins: {
-                [K in keyof CMixins]: CMixins[K] extends IClass<infer T> ? T : never
-            } = Mixins.map(function <T extends unknown>(Mixin: IClass<T>, index: number): T {
+            const mixins = Mixins.map(function (Mixin, index) {
                 return Reflect.construct(Mixin, argsMixins[index], self.constructor);
             });
 

